### PR TITLE
TextPanel: Fixes issue when interpolation of variables stops working

### LINKED ISSUE #26083: global built-in variables: __dashboard and __dashboard.name are not available in the text panel
I want to use the relatively new global variable __dashboard to open git hub issues.
When I try to use the variable in the panel it is not being replaced, I've tried both the `$` and the `[[]]` substitution none worked

How I just reproduce it:

* Edit a text panel and add ${__dashboard.name} in the text - Should work
* Exit the edit panel
* Edit the same panel again, remove and add the ${__dashboard.name} now it does not work

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
I couldn't really repro the issue as described in issue #26083. But I could get the interpolation of variables to stop working as we were transforming the content provided by the user to markdown before interpolating variables. This PR reverses the order, we interpolate variables before transforming the content to markdown.

I also updated the `marked` library and did some small refactoring, removing the remains of the `text` mode for instance.

**Which issue(s) this PR fixes**:
Fixes #26083 (couldn't repro the exact issue)

**Special notes for your reviewer**:

### CODE REVIEW

**Review by torkelo — APPROVED** [review]
Looks good!

**Comment by hugohaggmark** [comment]
@torkelo I needed to change the TextPanel from markdown to html because I guessed that we want to show the `raw` interpolated value and not a markdownified value. Let me know if this makes sense 🙏

**Comment by torkelo** [comment]
>  I needed to change the TextPanel from markdown to html because I guessed that we want to show the raw interpolated value and not a markdownified value. Let me know if this makes sense pray 

Not sure I understand. You mean in the test dashboard / e2e test?

File: devenv/dev-dashboards/feature-templating/global-variables-and-interpolation.json [hunk 1]
```diff
@@ -34,8 +34,8 @@
...
+        "mode": "html"
```
Comment (torkelo) [inline devenv/dev-dashboards/feature-templating/global-variables-and-interpolation.json:38]: Any reason for this change?  is not the above rendering to this (but with interpolated variables) anyway?

Comment (hugohaggmark) [reply devenv/dev-dashboards/feature-templating/global-variables-and-interpolation.json:38]: Sorry for being unclear. Let's see the upgraded markdown will transform an interpolated string like this `(A'A"A|BB\\B|CCC)` into this `(A'A"A|BB\B|CCC)`. 

Much like GitHub markdown renderer does with the same string below
(A'A"A|BB\\B|CCC) is in fact `(A'A"A|BB\\B|CCC)`
* `Server:regex` = (A'A"A|BB\\B|CCC)
* `Server:lucene` = ("A'A\"A" OR "BB\\B" OR "CCC")
* `Server:glob` = {A'A"A,BB\B,CCC}

So I thought that we wanted the e2e test to verify the correct string, the raw output from interpolation and not the markdownified version. Does this make sense? Hard to explain.

Just realized I should be able to surround the variables with ` and change it back to markdown.

Comment (hugohaggmark) [reply devenv/dev-dashboards/feature-templating/global-variables-and-interpolation.json]: Reverted back to markdown but I think there was a bug in the e2e test for the `:html` format so I had to change the test.

Comment (torkelo) [reply devenv/dev-dashboards/feature-templating/global-variables-and-interpolation.json]: ah, now I understand. now that interpolation happens before markdown markdown is impacting the interpolated value. 
In that case I agree html mode  is probably better :)

Comment (hugohaggmark) [reply devenv/dev-dashboards/feature-templating/global-variables-and-interpolation.json]: Hehehe I've already reverted back to markdown surrounding all values with `. Please see my latest commit :)

Comment (torkelo) [reply devenv/dev-dashboards/feature-templating/global-variables-and-interpolation.json]: ah, smart!

**Pull request merged**